sequencer/src/block.rs

```diff
@@ -178,6 +178,31 @@ impl Header {
     }
 }
 
+fn _validate_proposal(parent: &Header, proposal: &Header) -> anyhow::Result<SHA3MerkleTree> {
```

Comment:
Nice! Just a small nit: maybe we can make the merkle tree type a public type alias (something like `pub type BlockMerkleTree = SHA3MerkleTree`. Just so don't need to bake the hash function into the type name.